fix: bypass hostname verification for AMT 19+ when skip-amt-cert-check is enabled

sinchubhat opened the pull request:
* AMT 19+ certificates lack localhost in SANs, causing validation failures. Added VerifyConnection callback to bypass hostname verification when skipCertCheck is enabled. 
* Fixed WSMAN client setup in activation to use proper TLS config. 
* Added client certificate support for mutual TLS authentication. AMT 19+ devices require client certificates in TLS handshake for activation and post-activation WSMAN connections. 
* Added PTHI/MEI StartConfigurationHBased call for TLS-enforced initial activation. 
* Simplified orchestrator to pass provisioning cert via environment variables to all subprocess commands. After ACM activation on AMT 19+, the firmware requires client certificate authentication for all WSMAN connections.
* Implement automatic rollback on password verification failure after PTHI activation.
* Added retry logic and stabilization delays to resolve EOF errors during activation.
* Skip WSMAN setup for CCM activation (no admin account in pre-provisioning state).
* Reconnect with admin credentials after CCM HostBasedSetupService for CommitChanges.
* Fix orchestrator always passing --skip-amt-cert-check flag regardless of CLI input.
* Display helpful message instructing users to retry with --skip-amt-cert-check on rollback.
* Add WSMAN setup retry logic for post-activation AMT service stabilization. Fixes TLS configuration failures after CCM/ACM activation due to "interrupted system call" errors.

Addresses #1068, #1095

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ when skip-amt-cert-check is enabled

AMT 19+ certificates lack localhost in SANs, causing activation to fail
even with --skip-amt-cert-check. Added VerifyConnection callback to
properly bypass hostname validation when skipAMTCertCheck is true.
Also fixed WSMAN client setup in activation to use proper TLS config.

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ when skip-amt-cert-check is enabled

AMT 19+ certificates lack localhost in SANs, causing validation
failures. Added VerifyConnection callback to bypass hostname
verification when skipCertCheck is enabled.
Fixed WSMAN client setup in activation to use proper TLS config.
Added client certificate support for mutual TLS authentication.
AMT 19+ devices require client certificates in TLS handshake for
activation and post-activation WSMAN connections.
Added PTHI/MEI StartConfigurationHBased call for TLS-enforced
initial activation. Fixed orchestrator to propagate provisioning
cert and flags to spawned subprocesses.

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ when skip-amt-cert-check is enabled

Implement automatic rollback on password verification failure after PTHI activation.
Added retry logic and stabilization delays to resolve EOF errors during activation.
Skip WSMAN setup for CCM activation (no admin account in pre-provisioning state).
Reconnect with admin credentials after CCM HostBasedSetupService for CommitChanges.
Fix orchestrator always passing --skip-amt-cert-check flag regardless of CLI input.
Display helpful message instructing users to retry with --skip-amt-cert-check on rollback.

Addresses #1068

rsdmike reviewed (changes requested):
This has a lot of changes to simply support bypassing hostname verification. This PR needs to be broken down into smaller PRs for easier review. I'm not sure why all the configure commands have provisioning cert added to them as this should be available from a more global context.

sinchubhat committed:
Merge remote-tracking branch 'origin/main' into issue1068-rpc-go

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ when skip-amt-cert-check is enabled

After ACM activation on AMT 19+, client certificate authentication is
required for all WSMAN connections. Simplified code to pass provisioning
cert globally via environment variables to all subprocess commands instead
of individual CLI flags.

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ when skip-amt-cert-check is enabled

Fixed CodeQL security warning for InsecureSkipVerify
Replaced hardcoded control mode values (0/1/2) with named constants
Replaced hardcoded time delays with proper time.Duration constants
Refactored duplicate code patterns into helper functions

Addresses #1068

sinchubhat commented:
> This has a lot of changes to simply support bypassing hostname verification. This PR needs to be broken down into smaller PRs for easier review. I'm not sure why all the configure commands have provisioning cert added to them as this should be available from a more global context.

After ACM activation on AMT 19+, client certificate authentication is required for all WSMAN connections. Simplified code to pass provisioning cert globally via environment variables to all subprocess commands instead of individual CLI flags.
Note: this PR contains all the fixes for the errors I faced during activation/deactivation end-to-end on both AMT 19/21.

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ with skip-amt-cert-check

review comment fixes
lint error fixes
fixes for acm rollback when activated without --skip-amt-cert-check
display message for ccm when activated without --skip-amt-cert-check

Addresses #1068

sinchubhat committed:
Merge branch 'main' into issue1068-rpc-go

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ with skip-amt-cert-check

lint error fixes

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ with skip-amt-cert-check

lint error fixes

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ with skip-amt-cert-check

lint error fixes

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ with skip-amt-cert-check

Add WSMAN setup retry logic for post-activation AMT service
stabilization. Fixes TLS configuration failures after CCM/ACM
activation due to "interrupted system call" errors.

Addresses #1068

sinchubhat committed:
fix: bypass hostname verification for AMT 19+ with skip-amt-cert-check

lint error fixes

Addresses #1068

sinchubhat commented on internal/commands/activate/local.go:
```diff
@@ -630,22 +867,166 @@ func (service *LocalActivationService) activateACMLegacy() error {
+		// This will cause verification to fail on AMT 19+ where cert CN=AMT RCFG doesn't match localhost
+		// which is the expected secure behavior - user must explicitly skip cert checks
+		verifyTlsConfig.ServerName = "localhost"
+	}
```
Updated. Thanks.

sinchubhat committed:
Merge branch 'main' into issue1068-rpc-go

sinchubhat committed:
Merge branch 'main' into issue1068-rpc-go

sinchubhat commented:
Sure, thanks a lot for the detailed info.
